lib/ansible/modules/cloud/amazon/api_gateway.py

```diff
@@ -0,0 +1,205 @@
+'''
+
+RETURN = '''
+output:
+  description: the data returned by put_restapi in boto3
+  returned: success
+  type: dict
+  sample:
+    'data':
+      {
+          "id": "abc123321cba",
+          "name": "MY REST API",
+          "createdDate": 1484233401
+      }
+'''
+
+import os
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            api_id=dict(type='str', required=False),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            swagger_file=dict(type='str', default=None, aliases=['src', 'api_file']),
+            swagger=dict(type='str', default=None),
+            stage=dict(type='str', default=None),
+            deploy_desc=dict(type='str', default=None),
+        )
+    )
+
+    mutually_exclusive = [['swagger_file', 'swagger']]
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False,  )   # TODO !!!`
+
+    api_id = module.params.get('api_id')
+    state = module.params.get('state')
+    swagger_file = os.path.expanduser(module.params.get('swagger_file'))
+    swagger = module.params.get('swagger')
+    stage= module.params.get('stage')
+    deploy_desc= module.params.get('deploy_desc')
+
+#    check_mode = module.check_mode
+    changed = False
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    if not HAS_BOTOCORE:
+        module.fail_json(msg='Python module "botocore" is missing, please install it')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    try:
+        client = boto3_conn(module, conn_type='client', resource='apigateway',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e), exception=traceback.format_exc())
+
+    if not api_id:
+        desc="Incomplete API creation by ansible api_gateway module"
+        awsret=client.create_rest_api(name="ansible-temp-api", description=desc)
+        api_id=awsret["id"]
+
+    apidata=None
+    if swagger_file:
+        try:
+            with open(swagger_file) as f:
+                apidata=f.read()
+        except Exception as e:
+            module.fail_json(msg=str(e), exception=traceback.format_exc())
+    if swagger:
+        apidata=swagger
+
+    if apidata is None:
+        module.fail_json(msg='module error - failed to get API data')
+
+    try:
+        create_response=client.put_rest_api(body=apidata, restApiId=api_id, mode="overwrite")
+    except Exception as e:
```

Comment:
Instead of all exceptions, just catch boto3 ones so other things (out of memory, network errors, etc etc) show up differently.